tests/testthat/test-cnd-handlers.R

```diff
@@ -148,3 +148,33 @@ test_that("drop_global_handlers() works and is idempotent", {
   out <- Rscript(shQuote(c("--vanilla", "-e", code)))
   expect_equal(out$out, chr())
 })
+
+test_that("stackOverflowError are caught", {
+  handler <- function(cnd) handled <<- TRUE
+  overflow <- function() signal("", "stackOverflowError")
+
+  handled <- FALSE
+  try_call(overflow(), error = handler)
```

Comment:
```suggestion
  try_call(overflow(), error = ~ handled <<- TRUE)
```

Maybe? It's not much longer and makes it slightly easier to read